spec/app/domain/authentication/resource_restrictions/get_restriction_from_annotation_spec.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe(Authentication::ResourceRestrictions::GetRestrictionFromAnnotation) do
+  let(:authenticator_name) { 'authn-dummy' }
+  let(:service_name) { 'dummy_service' }
+  let(:restriction_name) { 'user_email' }
+  let(:general_annotation_name) { "#{authenticator_name}/#{restriction_name}" }
+  let(:service_specific_annotation_name) { "#{authenticator_name}/#{service_name}/#{restriction_name}" }
+
+  #  ____  _   _  ____    ____  ____  ___  ____  ___
+  # (_  _)( )_( )( ___)  (_  _)( ___)/ __)(_  _)/ __)
+  #   )(   ) _ (  )__)     )(   )__) \__ \  )(  \__ \
+  #  (__) (_) (_)(____)   (__) (____)(___/ (__) (___/
+
+  context "GetRestrictionFromAnnotation" do
+    context "Annotation is general" do
```

Comment:
what about tests:

- Annotation name without authenticator name
- Annotation name like {authenticator_name}/{authenticator_name}/
- Annotation name empty ? nil ? it is a possible input?
- Annotation name with more than 2 / like {something}/{something}/{something}/{something}
- Annotation name like with service id first {service_name}/#{authenticator_name}


those just examples, your are more familiar with the possible pitfalls in the code, try to challenge it

Reply:
Fixed